diff --git a/test/util/e2e.go b/test/util/e2e.go
index 5fd0e6600..992bdc49f 100644
--- a/test/util/e2e.go
+++ b/test/util/e2e.go
@@ -721,5 +721,10 @@ func CreateNamespaceFromPrefixWithLog(ctx context.Context, k8sClient client.Clie
 func CreateNamespaceFromObjectWithLog(ctx context.Context, k8sClient client.Client, ns *corev1.Namespace) *corev1.Namespace {
-	MustCreate(ctx, k8sClient, ns)
-	ginkgo.GinkgoLogr.Info("Created namespace", "namespace", ns.Name)
-	return ns
+	ginkgo.GinkgoLogr.Info("Created namespace", "namespace", ns)
+	if ns.Labels == nil {
+		ns.Labels = make(map[string]string)
+	}
+	ns.Labels["kueue.openshift.io/managed"] = "true"
+	gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	ginkgo.GinkgoLogr.Info(fmt.Sprintf("Created namespace: %s", ns.Name))
+	return ns
 }
@@ -757,2 +762,2 @@ func ExpectMetricsToBeAvailable(ctx context.Context, cfg *rest.Config, restClien
-	}, LongTimeout, Interval).Should(gomega.Succeed())
+	}, VeryLongTimeout, Interval).Should(gomega.Succeed())
 }
@@ -766,2 +771,2 @@ func ExpectMetricsNotToBeAvailable(ctx context.Context, cfg *rest.Config, restCl
-	}, LongTimeout, Interval).Should(gomega.Succeed())
+	}, VeryLongTimeout, Interval).Should(gomega.Succeed())
 }
